### PR TITLE
chore(integer): move comparator test out of it

### PR DESCRIPTION
**Pull request by tmontaigu**
Move the comparisons test (eq, ne, ge, gt, etc)
that were in the comparator module out of the comparator module.

This is so that in later commits will create test cases out of these tests so they can, like other unsigned tests be used to test other implementations of ServerKey

### CODE REVIEW

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs [hunk 1]
```diff
@@ -0,0 +1,1946 @@
...
+fn test_max_parallelized_256_bits(params: crate::shortint::ClassicPBSParameters, num_tests: usize) {
+    test_default_function(
+        params,
+        num_tests,
+        |comparator, lhs, rhs| comparator.max_parallelized(lhs, rhs),
+        std::cmp::max,
+    )
+}
+
+// macro_rules! create_parametrized_test{
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs:256]: comment

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs [hunk 2]
```diff
@@ -0,0 +1,1946 @@
...
+#[test]
+fn test_min_parallelized_256_bits_param_message_2_carry_2_ks_pbs() {
+    test_min_parallelized_256_bits(
+        crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS,
+        4,
+    )
+}
+
+// #[test]
+// fn test_min_parallelized_256_bits_param_message_3_carry_3_ks_pbs() {
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs:451]: commented out ? add a reason

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs [hunk 3]
```diff
@@ -0,0 +1,1946 @@
...
+#[test]
+fn test_max_parallelized_256_bits_param_message_2_carry_2_ks_pbs() {
+    test_max_parallelized_256_bits(
+        crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS,
+        4,
+    )
+}
+
+// #[test]
+// fn test_max_parallelized_256_bits_param_message_3_carry_3_ks_pbs() {
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs:527]: same thing

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs (hunk 2 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs:451]: They were already commented, its the same reason as for non unchecked comparisons in 3_3

Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs:439]: comment added

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs (hunk 3 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs:516]: added

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs]: removed

**Pull request merged**